backend/src/infra/face-auth-repo.ts

```diff
@@ -0,0 +1,63 @@
+import { err, ok, ResultAsync } from "neverthrow";
+import { FieldValue } from "firebase-admin/firestore";
+import { createId } from "@paralleldrive/cuid2";
+import { FaceAuthError } from "./face-auth-repo.error";
+import type { FirebaseApp } from "../firebase";
+import { FirestoreInternalError } from "./shared/firestore-error";
+import { CustomerId } from "../domain/customer";
+
+export type RegisterEmbedding = (
+  firebase: FirebaseApp
+) => (embedding: number[]) => ResultAsync<string, FirestoreInternalError>;
+
+export type AuthenticateFace = (
+  firebase: FirebaseApp
+) => (
+  embedding: number[]
+) => ResultAsync<string, FaceAuthError | FirestoreInternalError>;
```

Comment:
imo: ここの戻り値は `domain/customer.ts` の `CustomerId` でもいいかもですねー。実際に値を返すところでは as で string -> CustomerId のキャストをしちゃってOKです
```suggestion
export type RegisterEmbedding = (
  firebase: FirebaseApp
) => (embedding: number[]) => ResultAsync<CustomerId, FirestoreInternalError>;

export type AuthenticateFace = (
  firebase: FirebaseApp
) => (
  embedding: number[]
) => ResultAsync<CustomerId, FaceAuthError | FirestoreInternalError>;
```

Reply:
fixed: f19697494078c3b83c1ab8e76270206fb0bd7390